src/RA_httpthread.cpp

```diff
@@ -118,10 +118,12 @@ static void AppendIntegrationVersion(_Inout_ std::string& sUserAgent)
     sUserAgent.append(ra::StringPrintf("%d.%d.%d.%d", RA_INTEGRATION_VERSION_MAJOR,
                                        RA_INTEGRATION_VERSION_MINOR, RA_INTEGRATION_VERSION_REVISION,
                                        RA_INTEGRATION_VERSION_MODIFIED));
-
-    _CONSTANT_LOC posFound{ std::string_view{ RA_INTEGRATION_VERSION_PRODUCT }.find('-') };
-    constexpr std::string_view sAppend{ RA_INTEGRATION_VERSION_PRODUCT };
-    sUserAgent.append(sAppend, posFound);
+    
+    if constexpr (_CONSTANT_LOC pos{ std::string_view{ RA_INTEGRATION_VERSION_PRODUCT }.find('-') }; pos != std::string_view::npos)
+    {
+        constexpr std::string_view sAppend{ RA_INTEGRATION_VERSION_PRODUCT };
+        sUserAgent.append(sAppend, pos);
```

Comment:
If not on a branch, this line would throw an exception. So it's in an if statement.